code/headers/module_task_base.hpp

```diff
@@ -14,6 +14,9 @@ namespace r2d2::module_scheduler {
         std::array<rtos::waitable *, 4> waitables;
 
     public:
+        module_task_base_c() : comm(), waitables() {
```

Comment:
Instead of call the std::array default constructor, you could use `waitables.fill(nullptr);` to make sure the pointers are nullptr.

Reply:
fixed in https://github.com/R2D2-2019/module_scheduler/pull/2/commits/f59de34372675a5341add83c3051639e3a7de62d